### PR TITLE
fix: dotnet function provider fixes, package.json cleanup, add pkg refs

### PR DESCRIPTION
**Pull request by attilah**
*Description of changes:*

- dotnet function provider fixes and refactor
- add missing packages to dotnet provider
- package.json cleanup, standardization

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts [hunk 1]
```diff
@@ -1,53 +1,35 @@
...
+      hasRequiredDependencies: false,
+      errorMessage: `Unable to find ${executableName} version ${currentSupportedVersion} on the path.`,
+    };
+  }
+
+  const result = execa.sync(executableName, ['--version']);
+  const versionString = result.stdout;
+
+  if (result.exitCode !== 0) {
+    throw new Error(`${executableName} failed, exit code was ${result.exitCode}`);
```
Comment (edwardfoyle) [inline packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts:22]: should this be a throw or return with errorMessage?

File: packages/amplify-dotnet-function-runtime-provider/src/utils/invoke.ts [hunk 2]
```diff
@@ -1,82 +1,58 @@
...
-      stdio: ['pipe', 'pipe', process.stderr],
-    });
-
-    invokeCommand.stdin.setDefaultEncoding('utf-8');
-    invokeCommand.stdin.write(request.event);
-    invokeCommand.stdin.end();
+  const result = execa.sync(executableName, [shimExecutablePath, request.handler], {
+    cwd: request.srcRoot,
+    env: {
+      ...process.env,
```
Comment (edwardfoyle) [inline packages/amplify-dotnet-function-runtime-provider/src/utils/invoke.ts:24]: do we want to be passing the cli's env to the lambda?

Comment (attilah) [reply packages/amplify-dotnet-function-runtime-provider/src/utils/invoke.ts:24]: you can have extra things there, it is fine. no harm can be made.

File: packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts (hunk 1 above)
Comment (attilah) [reply packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts:22]: throw. it is an error in this case.

**Pull request merged**